src/see.cpp

```diff
@@ -0,0 +1,278 @@
+#include <Rcpp.h>
+#include <math.h>
+using namespace Rcpp;
+
+//' Two-line Intersection
+//' 
+//' Calculate the intersection between two lines.
+//' 
+//' The first line is defined by \code{P1} and \code{P1}, the second by 
+//' \code{P3} \code{P4}. Returns \code{Inf} when the lines are parallel. If
+//' \code{interior_only} is \code{TRUE}, returns \code{NA} if the intersection
+//' is not within the plane spanning between the two lines.
+//' 
+//' @param P1,P2,P3,P4 Numeric vector with x- and y-coordinates.
+//' @param interior.only,interior_only Logical scalar indicating whether the intersection must
+//' be within the spans of the two lines.
+//' 
+//' @returns A numeric vector with x- and y-coordinates defining the
+//' intersecting point between the two lines.
+//' 
+//' @source Weisstein, Eric W. "Line-Line Intersection.
+//' "From MathWorld--A Wolfram Web Resource.
+//' \url{http://mathworld.wolfram.com/Line-LineIntersection.html}
+//' @author David Sterratt

```

Comment:
Isn't it too much to add David Sterratt as 'author'? This way it looks like that the code below was taken from David Sterratt. I looked it up, and this citation does not contain direct code, but rather an explanation of a well known matrix multiplication procedure. Something that you may find also on wikipedia (see: https://en.wikipedia.org/wiki/Line%E2%80%93line_intersection )

Reply:
That is a good point. I just copied this information from the original R code, but I think it is not necessary.